clients/drcachesim/tracer/raw2trace.h

```diff
@@ -777,18 +777,24 @@ template <typename T> class trace_converter_t {
                     buf->size, (ptr_uint_t)buf->addr);
         *buf_in = ++buf;
         // To avoid having to backtrack later, we read ahead to see whether this memref
-        // faulted.  There's a footer so this should always succeed.
+        // faulted.
+        // It is possible that this is the last entry in a thread buffer. In that case:
+        // 1) There's at least one more thread buffer. That thread buffer would start
+        //    with a timestamp, which means the check for interrupted bb would be false.
+        // 2) There's no other thread buffer. This is the last one. In that case, we
+        //    should have had a footer, and if we don't, we have a malformed trace,
+        //    which we can detect irrespective of this test.
```

Comment:
This long comment doesn't seem necessary: I would remove it as I'm not sure what it's adding?  What is needed is to explain that a fault could never happen across buffers, because there is no buffer dump point on an app fault event, so we're sure the final memref in a buffer was not interrupted.

Reply:
OK - I'll add it in the CL description, though, because I want to have a clear explanation as to why the change is sound.

Reply:
I don't understand why `1)` is there: why do we need to prove anything about the interrupted check?